File: Level 1/시저 암호/solution.js
function solution(s, n) {
  const ALPHABET_LENGTH = 26;

  return [...s]
    .map((char) => {
      if (char === " ") {
        return char;
      }

      const charCode = char.charCodeAt();
      const charCodePlusN = charCode + n;
      const isLowerCase = charCode >= 65 && charCode <= 90;

      if (isLowerCase) {
        return charCodePlusN > 90
          ? String.fromCharCode(charCodePlusN - ALPHABET_LENGTH)
          : String.fromCharCode(charCodePlusN);
      }

      return charCodePlusN > 122
        ? String.fromCharCode(charCodePlusN - ALPHABET_LENGTH)
        : String.fromCharCode(charCodePlusN);
    })
    .join("");
}

File: Level 1/시저 암호/solution.py
def solution(s, n):
    return "".join(
        [
            (
                chr((ord(char) - ord("a") + n) % 26 + ord("a"))
                if char.islower()
                else (
                    chr((ord(char) - ord("A") + n) % 26 + ord("A"))
                    if char.isupper()
                    else char
                )
            )
            for char in s
        ]
    )
